### ISSUE TITLE
WISH: as string! should work on binary!

### ISSUE THREAD

**Issue by meijeru** [red/red#4131]
`to-string! <binary>` uses utf-8 decoding, which is fine for most purposes. However, when one wants to decode a binary value that is encoded using ISO-8859-1 or Windows-1252, and this is quite common in e-mail transmission, one should like to use _direct_ conversion of each byte into an 8-bit character. The mechanism is there: `as` allows `binary! -> string!` in its type-spec, but the implementation refuses this particular case, for no apparent reason.

**Comment by 9214** [comment]
I think `as binary! <string>` might also be useful.

**Comment by meijeru** [comment]
"_I think as binary! <string> might also be useful._" Careful: strings are stored with either 8, 16 or 32 bits per character, in an implementation-dependent way. Since `as binary!`  does not convert to UTF-8 it will leave the bit sequences intact, and the user may not be able to determine from the binary result what the bit-size per character actually is.

**Comment by 9214** [comment]
`as binary!` simply gives you a raw view into `string!`s buffer, it's up to the user to keep tabs on the encoding.

There's at least one use-case for that, related to matching strings by datatypes in Parse (`fast-lexer` branch), which, because of the lexer's byte-oriented nature, requires the input string to be converted to `binary!` first. `to` suffices, but it copies the content.

**Comment by dockimbel** [comment]
Encoding/decoding different locale like ISO-8859-1 is planned to be handled by the codec system.

**Comment by 9214** [comment]
Another use-case that I just stumbled upon: analyzing binary data.
```red
>> as string! find read/binary %binary.exe "deflate"
== {deflate 1.1.4 Copyright 1995-2002 Jean-loup Gailly ^@^@^@^@^@^@^@^@^@^@^@^@<ÀG^@^D^@^D^@^H^@^D^@ÁG^@^D^@^E^@^P^@^H^@ÁG^@^D^@^F^...
```

Without `as` I would have to `copy/part` the `binary!` first (without knowing in advance how much data to copy, and loosing the ability to navigate the original data with series actions) and then resort to `to`:

```red
>> to string! copy/part find read/binary %binary.exe "deflate" 50
== {deflate 1.1.4 Copyright 1995-2002 Jean-loup Gailly}
```

**Comment by hiiamboris** [comment]
Can't you just `parse read/binary %binary.exe [to "deflate" copy s to #"^@"] to string! s`?

**Comment by dockimbel** [comment]
> Another use-case that I just stumbled upon: analyzing binary data.

String! has a specfic internal encoding (UCS1, UCS2 or UCS4). If a string! is constructed from a binary! with unknown encoding, how can be possible to garantee that the string is properly encoded?

For example, what is `as string! {FFFF}` supposed to return? If the `unit` series field is set to 1 (which is the case because the series buffer is shared), then what `FF` means in UCS1?

**Comment by 9214** [comment]
@dockimbel right. I argued [recently](https://github.com/red/red/pull/4363#issuecomment-701354177) that only `as binary! <string>` case should remain. OTOH, if such `binary!` buffer is modified, then the resulting string will likely get corrupted.

**Comment by dockimbel** [comment]
> OTOH, if such binary! buffer is modified, then the resulting string will likely get corrupted.

We can't allow that to happen.
